backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VmAnalyzer.java

```diff
@@ -849,7 +849,7 @@ private void logVmDown() {
      * 3. Otherwise, the VM went down unexpectedly
      */
     private void proceedDisappearedVm() {
-        if (System.nanoTime() - getVmManager().getPowerOffTimeout() < 0) {
+        if (!getVmManager().isColdReboot() && System.nanoTime() - getVmManager().getPowerOffTimeout() < 0) {
```

Comment:
did you see this happening, that the vm stopped being reported from the host, in this flow? this shouldn't happen since the vm shuts down (rather than powered off) during cold-reboot. the only case I can see this happening is when we get the vm in down state, destroy it but not able to process the change in that monitoring cycle and so we get it as a "disappeared vm" in the next monitoring cycle so it's not impossible but very unlikely (the former monitoring cycle should fail before calling setColdRebootFlag) - if I'm right then let's document it so it should be clear that this is an handling of such corner case